cwms-data-api/src/test/java/cwms/cda/data/dao/StreamReachDaoTestIT.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2024 Hydrologic Engineering Center
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package cwms.cda.data.dao;
+
+import cwms.cda.api.DataApiTestIT;
+import cwms.cda.api.enums.Nation;
+import cwms.cda.api.errors.NotFoundException;
+import static cwms.cda.data.dao.DaoTest.getDslContext;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.Location;
+import cwms.cda.data.dto.stream.Bank;
+import cwms.cda.data.dto.stream.Stream;
+import cwms.cda.data.dto.stream.StreamLocation;
+import cwms.cda.data.dto.stream.StreamLocationNode;
+import cwms.cda.data.dto.stream.StreamNode;
+import cwms.cda.data.dto.stream.StreamReach;
+import cwms.cda.helpers.DTOMatch;
+import fixtures.CwmsDataApiSetupCallback;
+import fixtures.TestAccounts;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import mil.army.usace.hec.test.database.CwmsDatabaseContainer;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.AfterAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("integration")
+final class StreamReachDaoTestIT extends DataApiTestIT {
+
+    private static final String OFFICE_ID = TestAccounts.KeyUser.SWT_NORMAL.getOperatingOffice();
+    private static final List<Location> LOCATIONS_CREATED = new ArrayList<>();
```

Comment:
DataApiTestIT already contains a static final `List<Location>`
This should be used as it centralizes the logic for test location create/failure/deletion and reduces duplication.

Reply:
Was an easy enough to change to go ahead and just update now to use that existing list.